internal/gps/source_manager.go

```diff
@@ -28,6 +28,26 @@ import (
 // Used to compute a friendly filepath from a URL-shaped input.
 var sanitizer = strings.NewReplacer("-", "--", ":", "-", "/", "-", "+", "-")
 
+type Locker interface {
```

Comment:
shouldn't need to be exported. if there is a reason to, though, it'll need a comment.